src/Common/src/Interop/Linux/cgroups/Interop.cgroups.cs

```diff
@@ -79,6 +85,16 @@ private static bool TryReadMemoryValueFromFile(string path, out ulong result)
                         result = checked(ulongValue * multiplier);
                         return true;
                     }
+
+                    var encoding = new UTF8Encoding();
+                    string value = encoding.GetString(bytes);
```

Comment:
The only difference between `new UTF8Encoding()` and `Encoding.UTF8` is whether GetPreamble() returns empty or non-empty.  Since this isn't calling GetPreamble, or using a TextWriter, presumably it can use the cached object instead of allocating a new one?

Reply:
Also, maybe we want to just save that to a static initializer span and compare the bytes (if it's guaranteed to be UTF-8/ASCII).

Unless this is all one-time startup code, in which case maybe it's not worth it.

Reply:
The call chain is:

- `Interop.cgroups.TryGetMemoryLimit`
- `Process.GetWorkingSetLimits`
- `Process.MaxWorkingSet.get`

It seems to me that `MaxWorkingSet.get` can only call this method one-time: https://github.com/dotnet/corefx/blob/c265373119bc0449d347862862b43de35a95b186/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L262-L273